### DISCUSSION TITLE
rdkit can't read mol2 file generated by openbabel

### DISCUSSION THREAD

**Discussion by kexul** [rdkit/rdkit#4061] [Q&A] [answered]
I had a mol2 file converted by openbabel using the following code: 
```
obabel -:"Cn1ccnc1-c1cccc(Oc2cc(C(=O)O)cc(Oc3cc(C(=N)N)ccc3O)n2)c1" -O ligand.mol -p 7 

obabel ligand.mol -O ligand.mol2 --gen3d --best -xu
```
the `ligand.mol` can be read by rdkit, but the `ligand.mol2` file can't be read by rdkit using the `MolFromMol2File()` function, which raised an error: 
```
[15:11:52] Can't kekulize mol.  Unkekulized atoms: 1 2 3 4 5

[15:11:52] sanitise [15:11:52] *****: 
````
Can anyone help? Thanks!

**Comment by kexul** [comment]
Ok, It seems like a bug of rdkit, I've found another issue refer to this: https://github.com/rdkit/rdkit/issues/3985

**Comment by greglandrum** [comment, marked as answer]
As the documentation says - https://www.rdkit.org/docs/source/rdkit.Chem.rdmolfiles.html#rdkit.Chem.rdmolfiles.MolFromMol2Block - the RDKit expects the atom types and semantics from corina.

It's not terribly surprising that it won't parse an openbabel mol2 file.

If you can, the easiest fix is probably to have openbabel give you an SDF file instead of a mol2 file.

**Reply by kexul** [reply]
Thanks for your help @greglandrum ! It seems like SDF is a better supported file type by rdkit, I'll keep that in mind.

**Reply by greglandrum** [reply]
you're welcome. I think you will find that SDF is better supported in general.
